### PR TITLE
Mention the behavior of RouteAttribute

### PR DESCRIPTION
**Pull request by miloszkukla**
It's worth to mention the implication of using RouteAttribute.

### CODE REVIEW

**Comment by Rick-Anderson** [comment]
@rynowak  30 second review of nice improvement. Go from:

 When building a REST API, it's rare that you will want to use `[Route(...)]` on an action method.

To:
 When building a REST API, it's rare that you will want to use `[Route(...)]` on an action method **as it makes the action accept all HTTP methods.**

**Comment by Rick-Anderson** [comment]
Fixes #15154

**Comment by Rick-Anderson** [comment]
@dougbu  can you recommend someone besides @rynowak  (who's busy) to do this 30 second routing review?

**Comment by dougbu** [comment]
@JamesNK

File: aspnetcore/mvc/controllers/routing.md [hunk 1]
```diff
@@ -281,7 +281,7 @@ public IActionResult CreateProduct(...)
 For a URL path like `/products` the `ProductsApi.ListProducts` action will be executed when the HTTP verb is `GET` and `ProductsApi.CreateProduct` will be executed when the HTTP verb is `POST`. Attribute routing first matches the URL against the set of route templates defined by route attributes. Once a route template matches, `IActionConstraint` constraints are applied to determine which actions can be executed.
 
 > [!TIP]
-> When building a REST API, it's rare that you will want to use `[Route(...)]` on an action method. It's better to use the more specific `Http*Verb*Attributes` to be precise about what your API supports. Clients of REST APIs are expected to know what paths and HTTP verbs map to specific logical operations.
+> When building a REST API, it's rare that you will want to use `[Route(...)]` on an action method as it makes the action accept all HTTP methods. It's better to use the more specific `Http*Verb*Attributes` to be precise about what your API supports. Clients of REST APIs are expected to know what paths and HTTP verbs map to specific logical operations.
```
Comment (JamesNK) [inline aspnetcore/mvc/controllers/routing.md]: Minor change to make it slightly more technically correct (by default a route is open to all verbs. `[Route]` isn't making it accept all, it is the lack of a constraint).

```suggestion
> When building a REST API, it's rare that you will want to use `[Route(...)]` on an action method as the action will accept all HTTP methods. It's better to use the more specific `Http*Verb*Attributes` to be precise about what your API supports. Clients of REST APIs are expected to know what paths and HTTP verbs map to specific logical operations.
```

**Pull request merged**